code/modules/shuttle/navigation_computer.dm

```diff
@@ -227,9 +311,14 @@
 		if(hidden_turf_info)
 			. = SHUTTLE_DOCKER_BLOCKED_BY_HIDDEN_PORT
 
-	if(space_turfs_only)
+	if(whitelist_turfs && whitelist_turfs.len)
```

Comment:
`if(length(whitelist_turfs))`

If you're going to do a for loop over it anyway though the loop wont run if it's empty. This check is unnecessary.

Replace this check and loop with a typecache of allowed types instead